server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/VariableDefinitionUtil.java

```diff
@@ -108,11 +108,19 @@ public class VariableDefinitionUtil {
    * @return the list of errors
    */
   public List<SyntaxError> processNodeWithVariableDefinitions(Node node) {
-    Deque<VariableDefinitionNode> variableDefinitionNodes =
-        node.getChildren().stream()
+    Deque<VariableDefinitionNode> variableDefinitionNodes = new LinkedList<>();
+    node.getChildren().forEach(n -> {
+      if (n.getNodeType() == NodeType.VARIABLE_DEFINITION) {
+        variableDefinitionNodes.add((VariableDefinitionNode) n);
+      } else if (n.getNodeType() == NodeType.COPY) {
```

Comment:
Nice idea, but you didn't delete `VARIABLE_DEFINITION` nodes under the `COPY` node.
It will cause memory leack.

Reply:
Fixed